staging/src/k8s.io/client-go/util/workqueue/queue_test.go

```diff
@@ -198,3 +200,81 @@ func TestFinishShutDownUntilAllItemsAreDone(t *testing.T) {
 
 	finishedWG.Wait()
 }
+
+func TestForceShutDownIgnoringIfAllItemsAreDone(t *testing.T) {
+
+	proc, err := os.FindProcess(os.Getpid())
+	if err != nil {
+		t.Error(err)
+	}
+
+	firstSignalProcessed := setupTestSignalHandler()
+
+	q := workqueue.New()
+
+	q.Add("foo")
+	q.Add("bar")
+
+	firstItem, _ := q.Get()
+	secondItem, _ := q.Get()
+
+	finishedWG := sync.WaitGroup{}
+	finishedWG.Add(1)
+	go func() {
+		defer finishedWG.Done()
+		q.ShutDown()
+	}()
+
+	// Add as many items as possible until the queue is marked "ShuttingDown"
+	deltaItems := 0
+	shuttingDown := false
+	for !shuttingDown {
+		q.Add(deltaItems)
+		_, shuttingDown = q.Get()
+		deltaItems++
+	}
+
+	// We know we are shutting down now and cannot add new items, so mark all
+	// items we added until we started shutting down, as done
+	for i := 0; i <= deltaItems; i++ {
+		q.Done(i)
+	}
+
+	// Only send os.Interrupt here, as to make sure that the tests can be
+	// executed OS agnostically
+	if err := proc.Signal(os.Interrupt); err != nil {
```

Comment:
From the doc of `os.Interrupt`
```
On Windows, sending os.Interrupt to a process with os.Process.Signal is not implemented; it will return an error instead of sending a signal.
```
question: does Kubernetes run its tests on Windows? This test will not pass in that case.